packages/core/src/animation/AnimatorStateMachine.ts

```diff
@@ -71,4 +87,70 @@ export class AnimatorStateMachine {
     }
     return name;
   }
+
+  /**
+   * Add an entry transition.
+   * @param transition - The transition
+   */
+  addEntryStateTransition(transition: AnimatorStateTransition): AnimatorStateTransition;
+  /**
+   * Add an entry transition to the destination state.
+   * @param animatorState - The destination state
+   */
+
+  addEntryStateTransition(animatorState: AnimatorState): AnimatorStateTransition;
+
+  addEntryStateTransition(transitionOrAnimatorState: AnimatorStateTransition | AnimatorState): AnimatorStateTransition {
+    let transition: AnimatorStateTransition | AnimatorState;
+    if (transitionOrAnimatorState instanceof AnimatorState) {
+      transition = new AnimatorStateTransition();
+      transition.destinationState = transitionOrAnimatorState;
+    } else {
+      transition = transitionOrAnimatorState;
+    }
+    this._entryTransitions.push(transition);
+    return transition;
+  }
+
+  /**
+   * Remove an entry transition.
+   * @param transition - The transition
+   */
+  removeEntryStateTransition(transition: AnimatorStateTransition): void {
+    const index = this._entryTransitions.indexOf(transition);
+    index !== -1 && this._entryTransitions.splice(index, 1);
+  }
+
+  /**
+   * Add an any transition.
+   * @param transition - The transition
+   */
+  addAnyStateTransition(transition: AnimatorStateTransition): AnimatorStateTransition;
+  /**
+   * Add an any transition to the destination state.
+   * @param animatorState - The destination state
+   */
+
+  addAnyStateTransition(animatorState: AnimatorState): AnimatorStateTransition;
+
+  addAnyStateTransition(transitionOrAnimatorState: AnimatorStateTransition | AnimatorState): AnimatorStateTransition {
+    let transition: AnimatorStateTransition | AnimatorState;
+    if (transitionOrAnimatorState instanceof AnimatorState) {
+      transition = new AnimatorStateTransition();
+      transition.destinationState = transitionOrAnimatorState;
+    } else {
+      transition = transitionOrAnimatorState;
+    }
+    this._anyStateTransitions.push(transition);
+    return transition;
+  }
```

Comment:
Can warp a func shared by `EntryState` and `AnyState`

Reply:
done